prdoc/pr_5036.prdoc

```diff
@@ -0,0 +1,16 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: "[pallet_contracts] Modify the storage host function benchmarks to be run on an unbalanced storage trie"
+
+doc:
+  - audience: Runtime User
+    description: |
+      This PR modifies the storage host function benchmarks. Previously, they were run
+      on an empty storage trie. Now, they are run on an unbalanced storage trie
+      to reflect the worst-case scenario. This approach increases the storage host
+      function weights and decreases the probability of DoS attacks.
```

Comment:
Contracts can write in storage in a way that produces an unbalanced trie? I expected this to be impossible/prevented.

Reply:
All the keys for stored values are hashed using Blake2, so it is unlikely that this will happen in normal use.

Reply:
A contract can't create unbalanced tries because we hash the key from the contract. However, a contract could have a balanced but very big trie. The unbalanced trie is just use to simulate a deep trie for benchmarking.